server/artifacts/artifact_server.go

```diff
@@ -391,6 +391,10 @@ func (a *ArtifactServer) getArtifactAndDriver(ctx context.Context, nodeId, artif
 	// 4. Template defines ArchiveLocation
 
 	templateName := wf.Status.Nodes[nodeId].TemplateName
+	templateRef := wf.Status.Nodes[nodeId].TemplateRef
+	if templateName == "" && templateRef != nil {
+		templateName = templateRef.Template
+	}
```

Comment:
thanks for fixing this! One thing I noticed is that in workflow/util/util.go there's a method called getTemplateFromNode() which is basically performing this logic. Maybe you can call that?

Reply:
Another thing is that I somehow produced a scenario in which neither templateName nor templateRef is defined for a node (I'm referencing a ClusterWorkfowTemplate). Maybe I can provide you with the Workflow spec and ClusterWorkflowTemplate for that and we can make sure to handle that as best we can?

Reply:
Sure, you can pass it along!